src/main/java/com/chargerrobotics/sensors/GyroscopeSerial.java

```diff
@@ -1,65 +1,88 @@
 package com.chargerrobotics.sensors;
 
-import java.nio.ByteBuffer;
-
 import com.chargerrobotics.utils.ArduinoSerial;
 import com.chargerrobotics.utils.ArduinoSerialReceiver;
 import com.chargerrobotics.utils.ArduinoSerialReceiver.ArduinoListener;
+import java.nio.ByteBuffer;
 
 public class GyroscopeSerial extends ArduinoListener {
 
-	private volatile float x = 0;
-	private volatile float y = 0;
-	private volatile float z = 0;
-	
-	public GyroscopeSerial() {
-		ArduinoSerialReceiver.registerListener(this, (short)0xA02D);
-	}
+  private volatile float x = 0;
+  private volatile float y = 0;
+  private volatile float z = 0;
+
+  public GyroscopeSerial() {
+    ArduinoSerialReceiver.registerListener(this, (short) 0xA02D);
+  }
+
+  public void receiveData(ArduinoSerial serial, ByteBuffer buffer) {
+    if (buffer.remaining() >= 12) {
+      x = buffer.getFloat();
+      y = buffer.getFloat();
+      z = buffer.getFloat();
+    }
+  }
+
+  /**
+   * Gets the robot heading in degrees or {@code -1} if data is expired
+   *
+   * <p>Equivalent to {@link #getYaw()}
+   *
+   * @return Robot heading
+   */
+  public float getHeading() {
+    return getYaw();
+  }
+
+  /**
+   * Gets the robot heading in radians on the interval [-pi, pi) or {@code -1} if data is expired.
```

Comment:
Since this can return between `-pi` and `pi`, `-1` isn't really a very good value to represent "invalid". Two alternatives here:
- Pick a different value outside of the acceptable range.
- Throw an exception to be handled by anyone calling this method.